code/ui/components/src/components/Button/Button.tsx

```diff
@@ -1,29 +1,121 @@
-import type { FC, ComponentProps, ReactNode } from 'react';
-import React, { forwardRef } from 'react';
+import type { ButtonHTMLAttributes, SyntheticEvent } from 'react';
+import React, { forwardRef, useEffect, useState } from 'react';
 import { styled } from '@storybook/theming';
 import { darken, lighten, rgba, transparentize } from 'polished';
+import { Slot } from '@radix-ui/react-slot';
+
+export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  asChild?: boolean;
+  size?: 'small' | 'medium';
+  padding?: 'small' | 'medium';
+  variant?: 'outline' | 'solid' | 'ghost';
+  onClick?: (event: SyntheticEvent) => void;
+  disabled?: boolean;
+  active?: boolean;
+  onClickAnimation?: 'none' | 'rotate360' | 'glow' | 'jiggle';
 
-const ButtonWrapper = styled.button<{
+  /** @deprecated This API is not used anymore. It will be removed after 9.0. */
   isLink?: boolean;
+  /** @deprecated This API is not used anymore. It will be removed after 9.0. */
   primary?: boolean;
+  /** @deprecated This API is not used anymore. It will be removed after 9.0. */
   secondary?: boolean;
+  /** @deprecated This API is not used anymore. It will be removed after 9.0. */
   tertiary?: boolean;
+  /** @deprecated This API is not used anymore. It will be removed after 9.0. */
   gray?: boolean;
+  /** @deprecated This API is not used anymore. It will be removed after 9.0. */
   inForm?: boolean;
-  disabled?: boolean;
+  /** @deprecated This API is not used anymore. It will be removed after 9.0. */
   small?: boolean;
+  /** @deprecated This API is not used anymore. It will be removed after 9.0. */
   outline?: boolean;
+  /** @deprecated This API is not used anymore. It will be removed after 9.0. */
   containsIcon?: boolean;
-  children?: ReactNode;
-  href?: string;
-}>(
-  ({ small, theme }) => ({
+}
+
+export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
```

Comment:
Same question about `HTMLButtonElement` here

Reply:
We talked about it with @kasperpeulen and it looked right but I defer to him on what is best here.

Reply:
Yes, also inspired by shadcn.